File: src/main/java/org/hathitrust/htrc/tools/pairtreehelper/InvalidPairtreePathException.java
package org.hathitrust.htrc.tools.pairtreehelper;

import java.io.IOException;

public class InvalidPairtreePathException extends IOException {

    public InvalidPairtreePathException(String message) {
        super(message);
    }

    public InvalidPairtreePathException(String message, Throwable cause) {
        super(message, cause);
    }

    public InvalidPairtreePathException(Throwable cause) {
        super(cause);
    }
}

File: lib/src/main/scala/org/hathitrust/htrc/data/exceptions/InvalidPairtreePathException.scala
package org.hathitrust.htrc.data.exceptions

@SuppressWarnings(Array("org.wartremover.warts.Null"))
case class InvalidPairtreePathException(path: String, message: String = null, cause: Throwable = null)
  extends Exception(s"$path" + Option(message).map(msg => s": $msg").getOrElse(""), cause)